pandas/core/indexes/base.py

```diff
@@ -441,6 +442,8 @@ def __new__(
             return Index._simple_new(data, name=name)
 
         # index-like
+        elif isinstance(data, Index) and data._is_numeric_index() and dtype is None:
```

Comment:
umm what exactly is this catching?

Reply:
I've changed this a bit.

This captures the situation ``Index(NumericIndex(arr))` returns a `NumericIndex`. Without this the data would be converted to 64 bit and an `Int64Index`, `Float64Index` or `UInt64Index` would be returned.

I don't mind that conversion so much, but the test suite would have to be adapted so more work than the current implementation.